ncm-spma/src/main/perl/spma/yumng.pm

```diff
@@ -0,0 +1,810 @@
+		    if ( $repo_proxy eq 'no' ) {
+			while (@proxies) {
+			    my $prx = shift(@proxies);
+			    $prx .= ":$proxyport" if $spma->{proxyport};
+			    $url =~ s{(https?)://([^/]*)/}{$1://$prx/};
+			    $urls .= $url . ' ';
+			}
+		    } else {
+			$url .= ":$proxyport" if $spma->{proxyport};
+			$url =~ s,http?://,,;
+			$url =~ s,[^/]*/,,;
+			$url = $repo->{proxy} . $url;
+		    }
+		}
+	    }
+	}
+	if (!defined($urls)) {
+	    $urls = $url;
+	}
+        my $repofile = "[$repo->{name}]\nname=$repo->{name}\nbaseurl=$urls\nenabled=$repo->{enabled}\ngpgcheck=$repo->{gpgcheck}\n";
+        if ( defined( $repo->{mirrorlist} ) && $repo->{mirrorlist} ) {
+            $repofile = "[$repo->{name}]\nname=$repo->{name}\nmirrorlist=$url\nenabled=$repo->{enabled}\ngpgcheck=$repo->{gpgcheck}\n";
+        }
+        print $fh "# File generated by ", __PACKAGE__, ". Do not edit\n";
+        print $fh $repofile;
+        $fh->close();
+    }
+
+    # Preprocess required packages and separate version-locked
+    #    - also skip package dualities - e.g. both kernel and kernel-2.6.32-504.1.3.el6.x86_64
+    #      specified on yum install commandline will skip older version-locked package but
+    #      will install the latest what is undesired. Simply keep only version-locked variant.
+    my $pkgs               = $config->getElement(PKGS_TREE)->getTree();
+    my $wanted_pkgs        = Set::Scalar->new();
+    my $wanted_pkgs_locked = Set::Scalar->new();
+    my $found_spma         = 0;
+    my @pkl;
+    my @pkl_v;
+    my @pkl_a;
+
+    for my $name ( keys %$pkgs ) {
+        if ( !$found_spma && substr( unescape $name, 0, 8 ) eq 'ncm-spma' ) {
+            $found_spma = 1;
+        }
+        my $vra = $pkgs->{$name};
+        while ( my ( $vers, $a ) = each(%$vra) ) {
+            my $arches = $a->{arch};
+            if ( exists( $a->{repository} ) ) {
+                foreach my $arch (@$arches) {
+                    if ( $vers ne '_' ) {
+                        push ( @pkl_v, ( unescape $name) . ';' . ( unescape $vers) . '.' . $arch );
+                    } else {
+                        if ( $arch eq '_' ) {
+                            push ( @pkl, unescape $name . ';' );
+                        } else {
+                            push ( @pkl_a, unescape $name . ';' . $arch );
+                        }
+                    }
+                }
+            } else {
+                foreach my $arch ( keys %$arches ) {
+                    if ( $vers ne '_' ) {
+                        push ( @pkl_v, ( unescape $name) . ';' . ( unescape $vers) . '.' . $arch );
+                    } else {
+                        if ( $arch eq '_' ) {
+                            push ( @pkl, unescape $name . ';' );
+                        } else {
+                            push ( @pkl_a, unescape $name . ';' . $arch );
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    if ( !$found_spma ) {
```

Comment:
maybe use
```perl
if (! $pkgs->{escape('ncm-spma')}) {
  ...
}
```
before the above loop (and remove the logic looking for spma in the loop)